tst/regression/test_suites/particle_leapfrog/particle_leapfrog.py

```diff
@@ -50,7 +50,9 @@ def Analyse(self, parameters):
                 [0.12, 0.2, -0.3, 1.0, 1.0, 1.0],
             ]
         )
-        return (
-            structured_to_unstructured(final_data[["x", "y", "z", "vx", "vy", "vz"]])
-            == ref_data
-        ).all()
+        final_data = structured_to_unstructured(
+            final_data[["x", "y", "z", "vx", "vy", "vz"]]
+        )
+        if ref_data.shape != final_data.shape:
+            return False
```

Comment:
If we can fail because of different conditions, I suggest to expose that info:
```suggestion
        if ref_data.shape != final_data.shape:
            print("TEST FAIL: Mismatch between actual and reference data shape.")
            return False
```

Reply:
Good idea, thanks